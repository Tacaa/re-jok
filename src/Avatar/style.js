import styled, {css} from 'styled-components';

export const baseAvatar = css `
   display: flex;
   flex: none;
   align-self: center;
   align-items: center;
   justify-content: center;
   list-style: none;
   text-transform: uppercase;
   box-sizing: border-box;
   margin-right: 0.8rem;
   background: ${props => props.imgSrc
  ? "transparent"
  : "#bdbdbd"} ;
   color: #fafafa;
   font-size: ${props => props.theme.size[props.size].fontSize};
   line-height: ${props => props.theme.size[props.size].lineHeight};
   width: ${props => props.theme.size[props.size].width};
   height: ${props => props.theme.size[props.size].height};
   padding: 0;

`

export const StyledAvatar = styled.div `
  ${baseAvatar}

  border-radius: ${props => props.shape === "circle" ? '50%' : '.4rem'};

`
export const StyledAvatarImage = styled.img.attrs({
  src: props => props.imgSrc
    ? props.imgSrc
    : "",
  alt: props => props.imgAlt
    ? props.imgAlt
    : ""
})`

  border-radius: inherit;
  width: 100%;
  height: 100%;
  object-fit: cover;
  vertical-align: middle;
  border-style: none;
`

//=============== Avatar Group ==================
export const StyledAvatarGroup = styled.ul `
  list-style: none;
  display: inline-flex;
  margin-right: .8rem;
  margin-left: .8rem;
  pointer-events: auto;
  order: 1;
`

export const StyledAvatarItem = styled.li `
  margin-left: -.8rem;
  width: ${props => props.theme.size[props.size].width};
  height: ${props => props.theme.size[props.size].height};
  pointer-events: auto;
  box-shadow: rgb(255, 255, 255) 0px 0px 0px 2px;
  position: relative;
  border-radius: 50%;
  transition: transform 0.2s ease-in-out;
`
