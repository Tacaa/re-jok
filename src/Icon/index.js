// @flow
import * as React from 'react';

import FontAwesome from 'react-fontawesome';
import styled from 'styled-components'

const IconContainer = styled.div`
order: ${props => props.iconPosition === "left" ? 0 : 1};
color:  ${props => props.color ? props.theme[props.color].main : null};
display: inline-block;


`

type Props = {
  /**name of icon**/
  name: string,
  /** Size of icon shuold be string like lg, 2x, 3x v..v if you don't want use size you can custom with style**/
  size?: string,
  /**Override style of icon**/
  style?: Object,
  /**Add more class to icon**/
  className?: string,
  /** Set icon spin or not**/
  spin: boolean,
  /** Set color to icon but it just set color like error, warning,success.. i recomment you set in style or class**/
  color?: string,
  /** set postion of icon right or left**/
  iconPosition: 'left' | 'right',
}

const defaultProps = {
  spin: false,
  iconPosition: "left"
}


class Icon extends React.Component<Props> {
  static defaultProps = defaultProps;

  render(){
    const {
      iconPosition,
      name,
      size,
      style,
      className,
      spin,
      color,
    } = this.props;
    return (
      <IconContainer color={color} iconPosition={iconPosition}>
        <FontAwesome
         name={name}
         size={size}
         style={style}
         className={className}
         spin={spin}

        />
      </IconContainer>
    )
  }
}



export default Icon;
