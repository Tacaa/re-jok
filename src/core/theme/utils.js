import {
  defaultTheme
} from './index';


export const configTheme = (theme: Object) => {
  let jokTheme = {...defaultTheme, ...theme};
  return jokTheme;
}
