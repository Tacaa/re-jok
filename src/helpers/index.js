export const generalId = () => {
  return '_' + Math.random().toString(36).substr(2, 9);
}

//Default Function
export const noop = () => {}
