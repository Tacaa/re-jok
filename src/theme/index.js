// @flow
// export our theme to the provider
// This is globally available in styled-components when interpolating a function like so:
// ${(props) => props.theme}
// Or using import { withTheme } from 'styled-components';


export const defaultTheme = {
  size: {
    small: {
      fontSize: '1.2rem',
      padding: '0.5rem 0.9rem',
      "lineHeight": "1",
      "borderRadius": ".2rem",
      width: "2.4rem",
      height: "2.4rem"
    },
    default: {
      fontSize: '1.4rem',
      padding: '0.9rem 1.8rem',
      lineHeight: "1.5",
      borderRadius: ".4rem",
      width: "3.2rem",
      height: "3.2rem"
    },
    medium: {
      fontSize: '1.8rem',
      padding: '1.4rem 2.7rem',
      lineHeight: "1.5",
      borderRadius: ".6rem",
      width: "4rem",
      height: "4rem"

    },
    large: {
      fontSize: '2rem',
      padding: '1.8rem 3.6rem',
      lineHeight: "2",
      borderRadius: ".6rem",
      width: "4.8rem",
      height: "4.8rem"
    }
  },
  grid:{
    xs: 768,
    sm: 768,
    md: 992,
    lg: 1200,
    xl: 1920
  },
  text: {
    primary: "rgba(0, 0, 0, 0.85)",
    secondary: "rgba(0, 0, 0, 0.48)",
    hint: "rgba(0, 0, 0, 0.38)"
  },
  background: {
    paper: "#fff",
    default: "#fafafa"
  },
  typography: {
    title: {
      fontSize: "2rem",
      fontWeight: "600",
      lineHeight: "2.8rem"

    },
    subtitle: {
      fontSize: "1.6rem",
      fontWeight: "400",
      lineHeight: "2.4rem"
    }
  },
  primary: {
    main: "#1c8ffb",

  },
  secondary: {
    main: "#1b1c1d",
  },
  info: {
    main: "#1890ff"
  },
  error: {
    main: "#f5222d"
  },
  success: {
    main: "#52c41a"
  },
  warning: {
    main: "#faad14"
  },
  zIndex: {
    base: 1,
    background: 0,
    hidden: -1,
    card: 2,
    loading: 3,
    avatar: 3,
    form: 3,
    search: 3,
    badge: 100,
    modal: 1500,
  }
};

const changeTheme = (defaultTheme, theme)=>{
  let cloneTheme = {...defaultTheme};


  Object.keys(theme).forEach(function (key) {
    if ( 'object' === typeof theme[key] && !Array.isArray(theme[key]) ) {
    	changeTheme(cloneTheme[key], theme[key])
    }
    else if ( 'string' === typeof theme[key] || 'number' === typeof theme[key] ) {
      cloneTheme[key] = theme[key]
    }
  })
  return cloneTheme;
}

export const setTheme = (theme: Object) => {

   const newTheme = changeTheme(defaultTheme, theme);

   return newTheme;

}
