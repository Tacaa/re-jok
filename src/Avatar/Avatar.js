// @flow
import * as React from 'react'

import {Icon} from '../Icon';
import {StyledAvatar, StyledAvatarImage} from './style';

type Props = {
  /** Set size of avatar.The default size is default **/
  size: 'small' | 'default' | 'medium' | 'large',
  /** Set shape of avatar.The default shape is circle**/
  shape: 'circle' | 'square',
  /** Set incon in avatar**/
  icon?: string,
  /** Set alt attribute to image avatar**/
  imgAlt?: string,
  /** Set src attribute to image avatar**/
  imgSrc?: string,
  /** Add more class to avatar**/
  className?: string,
  /**Overrie or add style to root avatar**/
  style?: Object,
  /****/
  children?: any
}

const defaultProps = {
  size: 'default',
  shape: 'circle'
}

class Avatar extends React.Component<Props> {
  static defaultProps = defaultProps;
  render() {
    const {
      imgAlt,
      imgSrc,
      icon,
      ...rest
    } = this.props
    return (<StyledAvatar imgSrc={imgSrc} {...rest}>
      {
        imgSrc
          ? (<StyledAvatarImage imgAlt={imgAlt} imgSrc={imgSrc}/>)
          : icon
            ? (<Icon name={icon}/>)
            : this.props.children

      }

    </StyledAvatar>)
  }
}

export default Avatar;
