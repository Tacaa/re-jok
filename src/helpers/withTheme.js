import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import {defaultTheme} from '../theme';
import { ThemeProvider } from 'styled-components';
export function renderWithTheme(component) {

  return renderer.create(
    <ThemeProvider theme={defaultTheme}>
      {component}
    </ThemeProvider>
  );
};

export const shallowWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()
  return shallow(tree, { context })
};

export const mountWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()

  return mount(tree, {
    context,
    childContextTypes: ThemeProvider.childContextTypes
  })
};
