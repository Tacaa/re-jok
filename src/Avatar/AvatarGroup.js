// @flow
import * as React from 'react'
import Avatar from './Avatar'
import {StyledAvatarGroup, StyledAvatarItem} from './style'

type Props = {
  /** Children must be Avatar**/
  children: React.ChildrenArray < React.Element < typeof Avatar >>,
  /** size of avatar group must be like size of avatar**/
  size?: 'small' | 'default' | 'medium' | 'large'
}

const defaultProps = {
  size: "default"
}

class AvatarGroup extends React.Component<Props> {
  static defaultProps = defaultProps;
  render() {
    const {size, children} = this.props;
    return (<StyledAvatarGroup>
      {
        children && React.Children.count(children) > 0 && React.Children.map(children, (c, i) => (<StyledAvatarItem size={size} key={i}>
          {c}
        </StyledAvatarItem>))
      }
    </StyledAvatarGroup>)
  }
}

export default AvatarGroup
