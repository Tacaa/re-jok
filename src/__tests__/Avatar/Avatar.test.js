import React from 'react';
import Avatar from '../../Avatar/Avatar'
import {defaultTheme} from '../../theme'
import Icon from '../../Icon';
import { shallowWithTheme, mountWithTheme, renderWithTheme } from '../../helpers/withTheme';
import 'jest-enzyme'
import 'jest-styled-components'


describe('<Avatar />', () => {
  it('should render Avatar', () => {
    const tree = renderWithTheme(<Avatar>
      T
    </Avatar>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a Avatar with defaultProps', () => {
    const wrapper = shallowWithTheme(<Avatar/>, defaultTheme);
    expect(wrapper.props().size).toEqual('default');
    expect(wrapper.props().shape).toEqual('circle');

  });

  it('renders a circle avatar with size default', () => {
    const wrapper = mountWithTheme(<Avatar>T</Avatar>, defaultTheme);
    expect(wrapper.props().shape).toEqual("circle");
    expect(wrapper.text()).toEqual("T");

    const tree = renderWithTheme(<Avatar>T</Avatar>).toJSON();
    expect(tree).toHaveStyleRule('border-radius', '50%');
    expect(tree).toHaveStyleRule('font-size', '1.6rem');
    expect(tree).toHaveStyleRule('line-height', '1.5');
    expect(tree).toHaveStyleRule('width', '3.2rem');
    expect(tree).toHaveStyleRule('height', '3.2rem');


  });

  it('renders a square avatar with size default', () => {
    const wrapper = mountWithTheme(<Avatar shape="square">T</Avatar>, defaultTheme);
    expect(wrapper.props().shape).toEqual("square");
    expect(wrapper.text()).toEqual("T");

    const tree = renderWithTheme(<Avatar shape="square">T</Avatar>).toJSON();
    expect(tree).toHaveStyleRule('border-radius', '.4rem');
    expect(tree).toHaveStyleRule('font-size', '1.6rem');
    expect(tree).toHaveStyleRule('line-height', '1.5');
    expect(tree).toHaveStyleRule('width', '3.2rem');
    expect(tree).toHaveStyleRule('height', '3.2rem');


  });

  it('renders a icon avatar', () => {
    const wrapper = mountWithTheme(<Avatar icon="rocket">T</Avatar>, defaultTheme);
    expect(wrapper.props().icon).toEqual("rocket");

    expect(wrapper.find(Icon)).toHaveLength(1)

  });

  it('renders a image avatar', () => {
    const wrapper = mountWithTheme(<Avatar imgSrc="https://scontent.fhan2-1.fna.fbcdn.net/v/t1.0-1/c8.0.160.160/p160x160/13256326_1693157114242503_1952158310037985017_n.jpg?_nc_cat=0&_nc_eui2=AeHxGBHGerdp_tpPrKAqCXG_dT5O7zEHBxGSVtiqVcY-NJyK4OXo7i_pH-6207vmY_0yJ4iocMGa3IgmYhgSfCYPuAPyLQQBTnP-vPg-U_wZtw&oh=a421582f63736bf60c96f745a7ef9aa2&oe=5B843852"/>, defaultTheme);
    expect(wrapper.props().imgSrc).toEqual("https://scontent.fhan2-1.fna.fbcdn.net/v/t1.0-1/c8.0.160.160/p160x160/13256326_1693157114242503_1952158310037985017_n.jpg?_nc_cat=0&_nc_eui2=AeHxGBHGerdp_tpPrKAqCXG_dT5O7zEHBxGSVtiqVcY-NJyK4OXo7i_pH-6207vmY_0yJ4iocMGa3IgmYhgSfCYPuAPyLQQBTnP-vPg-U_wZtw&oh=a421582f63736bf60c96f745a7ef9aa2&oe=5B843852");

    expect(wrapper.find('img')).toHaveLength(1)
    expect(wrapper.find('img').prop('src')).toEqual("https://scontent.fhan2-1.fna.fbcdn.net/v/t1.0-1/c8.0.160.160/p160x160/13256326_1693157114242503_1952158310037985017_n.jpg?_nc_cat=0&_nc_eui2=AeHxGBHGerdp_tpPrKAqCXG_dT5O7zEHBxGSVtiqVcY-NJyK4OXo7i_pH-6207vmY_0yJ4iocMGa3IgmYhgSfCYPuAPyLQQBTnP-vPg-U_wZtw&oh=a421582f63736bf60c96f745a7ef9aa2&oe=5B843852")

  });



})
