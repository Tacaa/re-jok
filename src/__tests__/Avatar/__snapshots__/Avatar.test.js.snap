// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> should render Avatar 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  list-style: none;
  text-transform: uppercase;
  box-sizing: border-box;
  margin-right: 0.8rem;
  background: #bdbdbd;
  color: #fafafa;
  font-size: 1.6rem;
  line-height: 1.5;
  width: 3.2rem;
  height: 3.2rem;
  padding: 0;
  z-index: 3;
  border-radius: 50%;
}

<div
  className="c0"
  shape="circle"
  size="default"
>
  T
</div>
`;
